joomla/joomla-cms
# [4.0] Installation fixes

Successor PR to https://github.com/joomla/joomla-cms/pull/26250 

- Moves to promises per TODO
- Fixes bug where if token check failed the request still proceeded
- Simplifies installation logic
- Moves to promises/fetch in part of the installer (supported by all required browsers)

### Testing Instructions
Start a joomla 4 installation
In the "databse configuration"-tab enter incorrect database credentials
Click on the "Setup Database Connection" button
Now you should get a view, similar to the screenshot above
Now apply my patch
Retry the installation with incorrect database credentials
Now you should get a view where only the "Database Configuration"-Tab is shown

### Expected result
Only database tab shown for corrections

### Actual result
All installation Tabs are shown. (See screenshot above in PR #26250)

### Documentation Changes Required
None
